20090430133819_axtu.spec.diff

Index: /home/peter/sds/packages/axtu/1.5/tarball01/axtu/libs/classRpmEngine.cpp
===================================================================
--- /home/peter/sds/packages/axtu/1.5/tarball01/axtu/libs/classRpmEngine.cpp	(revision 7807)
+++ /home/peter/sds/packages/axtu/1.5/tarball01/axtu/libs/classRpmEngine.cpp	(working copy)
@@ -2542,7 +2542,12 @@
 	{
 		return -1;
 	}
-
+	string s_1;
+	s_1=(string)((const char *)key);
+	if(s_1.find("kernel")!= string::npos); //stupid ; here!!!
+	{
+        	nUpgrade=0;
+	}
 	if (nUpgrade)
 	{
 		if (CheckBlacklist((const char *)key) == true)
Index: /home/peter/sds/packages/axtu/1.5/axtu.spec
===================================================================
--- /home/peter/sds/packages/axtu/1.5/axtu.spec	(revision 7807)
+++ /home/peter/sds/packages/axtu/1.5/axtu.spec	(working copy)
@@ -27,7 +27,7 @@
 %endif
 
 NAME: axtu
-Release: 4%{?dist}
+Release: 5%{?dist}
 Summary: Asianux TSN Updater
 Group: Utilities/System
 License: GPL/LGPL
@@ -240,6 +240,9 @@
 %doc COPYING
 
 %changelog
+* Tue Apr 30 2009 Peter, Wang <peter.wang@asianux.com> 1.5.0-5%{?dist}
+- kmod-redcastle-PAE package can be installed now. (AXBUG#6012)
+
 * Wed Apr 29 2009 Lee, Jongwoo <oopslee@haansoft.com> 1.5.0-4%{dist}
 - fixed blacklist bug of obsoleted package. (AXBUG#6013, AXBUG#6174)
 - hided obsoleted package in install list. (AXBUG#6172, AXBUG#6173)


