20090518094507_axtu.spec.diff

Index: /home/peter/sds/packages/axtu/1.5/tarball01/axtu/libs/classRpmEngine.cpp
===================================================================
--- /home/peter/sds/packages/axtu/1.5/tarball01/axtu/libs/classRpmEngine.cpp	(revision 7848)
+++ /home/peter/sds/packages/axtu/1.5/tarball01/axtu/libs/classRpmEngine.cpp	(working copy)
@@ -782,10 +782,6 @@
 
 		/////////////
 		// Same name pass and find last version in RemoteServer's PkgSet.
-		set <structHeaderInfo, DereferenceLess>::iterator it_1;
-		it_1=itRemote;
-		string strEpoch_1;
-		bool flag_1=false;
 		while(strOldName2 == strNameR)
 		{
 			if(itRemote != m_setRemoteHeaderInfo.end()) itRemote++;
@@ -798,39 +794,16 @@
 			}
 			nCount++;
 			stripNVRA(itRemote->strNVRA, &strNameR, &strVerR, &strRelR, &strArchR);
-			strEpoch_1=itRemote->strEpoch;
-			if(!((strEpoch_1)==(it_1->strEpoch)))
-			{
 
-				if(!(strOldName2 != strNameR || (strOldName2 == strNameR && !CheckArch(strOldArch2 , strArchR))))
-
-				flag_1=true;
-			}
-           		if(flag_1)
-			{// now we need to concern about epoch
-
- 				if(strOldName2 != strNameR || (strOldName2 == strNameR && !CheckArch(strOldArch2 , strArchR)))
-				{
-					if((strEpoch_1)>(it_1->strEpoch))
-					it_1=itRemote;  //it latter epoch is bigger than early one, update it_1
-					itRemote=it_1;  
-					nCount--;
-					stripNVRA(itRemote->strNVRA, &strNameR, &strVerR, &strRelR, &strArchR);
-					break;
-				}
-			}
-			else
-			{  //now we have no epoch issued
-				if(strOldName2 != strNameR || (strOldName2 == strNameR && !CheckArch(strOldArch2 , strArchR)))
-				{   
+			if(strOldName2 != strNameR || (strOldName2 == strNameR && !CheckArch(strOldArch2 , strArchR)))
+			{   
 				itRemote--;
 				nCount--;
 				stripNVRA(itRemote->strNVRA, &strNameR, &strVerR, &strRelR, &strArchR);
 				break;
-				}
+			}
 
 
-			}
 		}
 		// arithmetic for Progress Value
 		if( !(nCount % 50) )
Index: /home/peter/sds/packages/axtu/1.5/tarball01/axtu/libs/classRpmEngine.h
===================================================================
--- /home/peter/sds/packages/axtu/1.5/tarball01/axtu/libs/classRpmEngine.h	(revision 7848)
+++ /home/peter/sds/packages/axtu/1.5/tarball01/axtu/libs/classRpmEngine.h	(working copy)
@@ -138,7 +138,18 @@
 		}		
 		else if( strArch1 == strArch2)
 		{
+			if((headerInfo1.strEpoch)<(headerInfo2.strEpoch))
+			{
+				nCompare=1;
+			}
+			else  if((headerInfo1.strEpoch)>(headerInfo2.strEpoch))
+			{
+				nCompare=-1;
+			}
+			else
+			{
 			nCompare = CompareVerRel2(strVer2, strRel2, strVer1, strRel1);
+			}
 			if (nCompare > 0 )
 			{
 				nResult = true;
Index: /home/peter/sds/packages/axtu/1.5/axtu.spec
===================================================================
--- /home/peter/sds/packages/axtu/1.5/axtu.spec	(revision 7848)
+++ /home/peter/sds/packages/axtu/1.5/axtu.spec	(working copy)
@@ -27,7 +27,7 @@
 %endif
 
 NAME: axtu
-Release: 7%{?dist}
+Release: 8%{?dist}
 Summary: Asianux TSN Updater
 Group: Utilities/System
 License: GPL/LGPL
@@ -242,6 +242,9 @@
 %doc COPYING
 
 %changelog
+* Thu May 18 2009 Peter, Wang <peter.wang@asianux.com> 1.5.0-8%{?dist}
+- irqbanlance will not show up on install list if it is installed already. (AXBUG#6263)
+
 * Thu May 14 2009 Peter, Wang <peter.wang@asianux.com> 1.5.0-7%{?dist}
 - fixed tui scroll bar problem. (AXBUG#4717)
 - fixed redcastle-app package conflicts problem. (AXBUG#5989)


