
20090619115358_axtu.spec.diff

Index: /home/peter/sds/packages/axtu/1.5/tarball01/axtu/libs/classRpmEngine.cpp
===================================================================
--- /home/peter/sds/packages/axtu/1.5/tarball01/axtu/libs/classRpmEngine.cpp	(revision 7957)
+++ /home/peter/sds/packages/axtu/1.5/tarball01/axtu/libs/classRpmEngine.cpp	(working copy)
@@ -2520,6 +2520,10 @@
 	if(s_1.find("kernel")!= string::npos)//if the package is a kernel install it
 	{
         	nUpgrade=0;
+		if(s_1.find("kernel-doc")!= string::npos||s_1.find("kernel-header")!= string::npos)
+		{
+			nUpgrade=1;
+		}
 	}
 	if (nUpgrade)
 	{
Index: /home/peter/sds/packages/axtu/1.5/axtu.spec
===================================================================
--- /home/peter/sds/packages/axtu/1.5/axtu.spec	(revision 7957)
+++ /home/peter/sds/packages/axtu/1.5/axtu.spec	(working copy)
@@ -27,7 +27,7 @@
 %endif
 
 NAME: axtu
-Release: 12%{?dist}
+Release: 13%{?dist}
 Summary: Asianux TSN Updater
 Group: Utilities/System
 License: GPL/LGPL
@@ -246,7 +246,11 @@
 %doc COPYING
 
 %changelog
-* Thu Jun 19 2009 Lee, Jongwoo <oopslee@haansoft.com> 1.5.0-12%{?dist}
+* Fri Jun 19 2009 Peter, Wang <peter.wang@asianux.com> 1.5.0-13%{?dist}
+- fixed "lots of packages can not be updated" problem.(AXBUG#6371)
+- fixed kernel-headers conflicts problem.(AXBUG#6362)
+
+* Thu Jun 18 2009 Lee, Jongwoo <oopslee@haansoft.com> 1.5.0-12%{?dist}
 - fixed upgrade bug of obsolete packages.(AXBUG#6369, AXBUG#6396) 
 
 * Wed Jun 10 2009 Lee, Jongwoo <oopslee@haansoft.com> 1.5.0-11%{?dist}


