20090415171714_axtu.spec.diff

Index: /home/peter/sds/packages/axtu/1.4/tarball01/axtu/libs/classRpmEngine.cpp
===================================================================
--- /home/peter/sds/packages/axtu/1.4/tarball01/axtu/libs/classRpmEngine.cpp	(revision 7718)
+++ /home/peter/sds/packages/axtu/1.4/tarball01/axtu/libs/classRpmEngine.cpp	(working copy)
@@ -2477,14 +2477,6 @@
 
 	//////////////////////////////////////////
 	//redcastle kernel module have to be installed by i option(install).
-	else if(strName == "redcastle-app")
-    {
-    	bInstall = true;
-    }
-	else if(strName == "redcastle-pam")
-    {
-    	bInstall = true;
-    }
 	else if(strName == "kmod-redcastle")
     {
     	bInstall = true;
Index: /home/peter/sds/packages/axtu/1.4/axtu.spec
===================================================================
--- /home/peter/sds/packages/axtu/1.4/axtu.spec	(revision 7718)
+++ /home/peter/sds/packages/axtu/1.4/axtu.spec	(working copy)
@@ -27,7 +27,7 @@
 %endif
 
 NAME: axtu
-Release: 4%{?dist}
+Release: 5%{?dist}
 Summary: Asianux TSN Updater
 Group: Utilities/System
 License: GPL/LGPL
@@ -238,7 +238,11 @@
 %doc COPYING
 
 %changelog
-* Wed Apr 10 2009 Peter, Wang <peter.wang@asianux.com> 1.4.16-4%{?dist}
+
+* Wed Apr 15 2009 Peter, Wang <peter.wang@asianux.com> 1.4.16-5%{?dist}
+- redcastle-app & redcastle-pam are now treated as common packages(AXBUG#5989 #5990)
+
+* Fri Apr 10 2009 Peter, Wang <peter.wang@asianux.com> 1.4.16-4%{?dist}
 - Added epoch check. (AXBUG#5965)
 
 * Wed Apr 08 2009 Lee, Jongwoo <oopslee@haansoft.com> 1.4.16-3%{?dist}


