diff -Nurp unchanged15/tarball01/axtu/gui/axtu/classGui.cpp 1.5/tarball01/axtu/gui/axtu/classGui.cpp
--- unchanged15/tarball01/axtu/gui/axtu/classGui.cpp	2009-06-14 08:40:10.000000000 +0800
+++ 1.5/tarball01/axtu/gui/axtu/classGui.cpp	2009-06-14 08:27:49.000000000 +0800
@@ -433,6 +433,16 @@ bool classGui::InitRpm()
 		}
 		
 		m_rpmEngine->SetCreateUpdateInstallListCallBack(CommonCallBack);
+//peter
+ if(m_rpmEngine->CreateUpdateInstallList()==111)
+{
+               m_Logger->WriteLog_char(ERROR_LOG, "classGUI", "inside1", NULL);
+               QMessageBox::information(this, staticTitle, tr("Newer version of rpm available.Plz update it first."));
+
+
+}
+
+//peter
 
 		if(!m_rpmEngine->CreateUpdateInstallList())
 		{	
diff -Nurp unchanged15/tarball01/axtu/libs/classRpmEngine.cpp 1.5/tarball01/axtu/libs/classRpmEngine.cpp
--- unchanged15/tarball01/axtu/libs/classRpmEngine.cpp	2009-06-14 08:43:03.000000000 +0800
+++ 1.5/tarball01/axtu/libs/classRpmEngine.cpp	2009-06-14 08:23:58.000000000 +0800
@@ -607,7 +607,7 @@ Creaet update list and install list by h
 */
 ///////////////////////////////////////////////////////////////////////////////////////////////////
 // Create update package list, install package list
-bool classRpmEngine::CreateUpdateInstallList()
+int classRpmEngine::CreateUpdateInstallList()
 {
 
 	m_nIncompatible = 0;
@@ -673,6 +673,7 @@ bool classRpmEngine::CreateUpdateInstall
 	int nCount = 0;
 	bool bFind;
 	bool bIsLoop=false;
+	int rpm_found=0;//peter add
 	structFileInfo fileInfo;
 
 	int nTotal = m_setRemoteHeaderInfo.size();
@@ -859,6 +860,16 @@ bool classRpmEngine::CreateUpdateInstall
 					} else {
 						fileInfo.bIncompatible = false;
 					}
+//peter
+                                       if(strNameR=="rpm")
+                                       {
+                                               m_Logger->WriteLog_char(ERROR_LOG, "strName:",strNameR.c_str(),NULL);
+                                               m_Logger->WriteLog_char(ERROR_LOG, "strVerR:",strVerR.c_str(),NULL);
+                                                m_Logger->WriteLog_char(ERROR_LOG, "strRelR:",strRelR.c_str(),NULL);
+                                                m_Logger->WriteLog_char(ERROR_LOG, "return","111",NULL);
+                                                 rpm_found=1;
+                                       }
+//peter-end
 
 
 					AddUpdateInstallList(&fileInfo,strNameR, \
@@ -940,6 +951,14 @@ bool classRpmEngine::CreateUpdateInstall
   // return rtv;
   // }
 
+//peter
+if(rpm_found)
+{
+
+return 111;
+}
+//peter-end
+
 
 	if(!bIsLoop){
 		ApplyBlacklistToUpdate();
diff -Nurp unchanged15/tarball01/axtu/libs/classRpmEngine.h 1.5/tarball01/axtu/libs/classRpmEngine.h
--- unchanged15/tarball01/axtu/libs/classRpmEngine.h	2009-06-14 08:42:26.000000000 +0800
+++ 1.5/tarball01/axtu/libs/classRpmEngine.h	2009-06-14 08:12:21.000000000 +0800
@@ -246,7 +246,7 @@ public:
 	int ReadHeaders();    // -------------------------------------------------- Callback
 	int OpenHeader(string strFilePath, int nIndex, bool bUpgrade);
 	int CompareHeaderInfo(vector <structHeaderInfo> vectorHeaderInfo, structHeaderInfo headerInfo);	
-	bool CreateUpdateInstallList();
+	int CreateUpdateInstallList();//peter
 				
 	void ClearAddedFiles();
 	
