--- elfutils.spec_old	2007-02-27 08:43:11.000000000 +0800
+++ elfutils.spec	2009-03-10 19:20:23.000000000 +0800
@@ -1,5 +1,5 @@
-%define eu_version 0.124
-%define eu_release 1
+%define eu_version 0.137
+%define eu_release 3
 
 %if %{?_with_compat:1}%{!?_with_compat:0}
 %define compat 1
@@ -7,7 +7,21 @@
 %define compat 0
 %endif
 
-Summary: A collection of utilities and DSOs to handle compiled objects.
+%if 0%{?fedora} >= 8
+%define scanf_has_m 1
+%endif
+%if 0%{?rhel} >= 6
+%define scanf_has_m 1
+%endif
+
+%if 0%{?fedora} >= 7
+%define separate_devel_static 1
+%endif
+%if 0%{?rhel} >= 6
+%define separate_devel_static 1
+%endif
+
+Summary: A collection of utilities and DSOs to handle compiled objects
 Name: elfutils
 Version: %{eu_version}
 %if !%{compat}
@@ -15,22 +29,17 @@
 %else
 Release: 0.%{eu_release}
 %endif
-License: GPL
+License: GPLv2 with exceptions
 Group: Development/Tools
-Source: elfutils-%{version}.tar.gz
+URL: https://fedorahosted.org/elfutils/
+Source: http://fedorahosted.org/releases/e/l/elfutils/%{name}-%{version}.tar.gz
 Patch1: elfutils-portability.patch
 Patch2: elfutils-robustify.patch
-Obsoletes: libelf libelf-devel
-Requires: elfutils-libelf = %{version}-%{release}
-Requires: elfutils-libs = %{version}-%{release}
-
-Patch0: elfutils-strip-copy-symtab.patch
-Source2: testfile16.symtab.bz2
-Source3: testfile16.symtab.debug.bz2
-
-# ExcludeArch: xxx
+Patch3: elfutils-0.137-fixes.patch
+Requires: elfutils-libelf-%{_arch} = %{version}-%{release}
+Requires: elfutils-libs-%{_arch} = %{version}-%{release}
 
-BuildRoot: %{_tmppath}/%{name}-root
+BuildRoot: %{_tmppath}/%{name}-%{version}-%{release}-root-%(%{__id_u} -n)
 BuildRequires: bison >= 1.875
 BuildRequires: flex >= 2.5.4a
 BuildRequires: bzip2
@@ -54,14 +63,10 @@
 
 
 %package libs
-Summary: Libraries to handle compiled objects.
+Summary: Libraries to handle compiled objects
 Group: Development/Tools
-License: GPL
-Requires: elfutils-libelf = %{version}-%{release}
-Conflicts: elfutils < %{version}-%{release}
-Conflicts: elfutils > %{version}-%{release}
-Conflicts: elfutils-devel < %{version}-%{release}
-Conflicts: elfutils-devel > %{version}-%{release}
+Provides: elfutils-libs-%{_arch} = %{version}-%{release}
+Requires: elfutils-libelf-%{_arch} = %{version}-%{release}
 
 %description libs
 The elfutils-libs package contains libraries which implement DWARF, ELF,
@@ -70,11 +75,14 @@
 other programs using these libraries.
 
 %package devel
-Summary: Development libraries to handle compiled objects.
+Summary: Development libraries to handle compiled objects
 Group: Development/Tools
-License: GPL
-Requires: elfutils-libs = %{version}-%{release}
-Requires: elfutils-libelf-devel = %{version}-%{release}
+Provides: elfutils-devel-%{_arch} = %{version}-%{release}
+Requires: elfutils-libs-%{_arch} = %{version}-%{release}
+Requires: elfutils-libelf-devel-%{_arch} = %{version}-%{release}
+%if !0%{?separate_devel_static}
+Requires: elfutils-devel-static-%{_arch} = %{version}-%{release}
+%endif
 
 %description devel
 The elfutils-devel package contains the libraries to create
@@ -83,15 +91,22 @@
 the DWARF debugging information.  libasm provides a programmable
 assembler interface.
 
+%package devel-static
+Summary: Static archives to handle compiled objects
+Group: Development/Tools
+Provides: elfutils-devel-static-%{_arch} = %{version}-%{release}
+Requires: elfutils-devel-%{_arch} = %{version}-%{release}
+Requires: elfutils-libelf-devel-static-%{_arch} = %{version}-%{release}
+
+%description devel-static
+The elfutils-devel-static package contains the static archives
+with the code to handle compiled objects.
+
 %package libelf
-Summary: Library to read and write ELF files.
+Summary: Library to read and write ELF files
 Group: Development/Tools
-Conflicts: elfutils < %{version}-%{release}
-Conflicts: elfutils > %{version}-%{release}
-Conflicts: elfutils-libs < %{version}-%{release}
-Conflicts: elfutils-libs > %{version}-%{release}
-Conflicts: elfutils-libelf-devel < %{version}-%{release}
-Conflicts: elfutils-libelf-devel > %{version}-%{release}
+Provides: elfutils-libelf-%{_arch} = %{version}-%{release}
+Obsoletes: libelf <= 0.8.2-2
 
 %description libelf
 The elfutils-libelf package provides a DSO which allows reading and
@@ -102,8 +117,12 @@
 %package libelf-devel
 Summary: Development support for libelf
 Group: Development/Tools
-Requires: elfutils-libelf = %{version}-%{release}
-Conflicts: libelf-devel
+Provides: elfutils-libelf-devel-%{_arch} = %{version}-%{release}
+Requires: elfutils-libelf-%{_arch} = %{version}-%{release}
+%if !0%{?separate_devel_static}
+Requires: elfutils-libelf-devel-static-%{_arch} = %{version}-%{release}
+%endif
+Obsoletes: libelf-devel <= 0.8.2-2
 
 %description libelf-devel
 The elfutils-libelf-devel package contains the libraries to create
@@ -111,21 +130,36 @@
 access the internals of the ELF object file format, so you can see the
 different sections of an ELF file.
 
+%package libelf-devel-static
+Summary: Static archive of libelf
+Group: Development/Tools
+Provides: elfutils-libelf-devel-static-%{_arch} = %{version}-%{release}
+Requires: elfutils-libelf-devel-%{_arch} = %{version}-%{release}
+
+%description libelf-devel-static
+The elfutils-libelf-static package contains the static archive
+for libelf.
+
 %prep
 %setup -q
 
-%patch0 -p1
-ln %{SOURCE2} %{SOURCE3} tests
+%if !0%{?scanf_has_m}
+sed -i.scanf-m -e 's/%m/%a/' tests/line2addr.c
+%endif
 
 %if %{compat}
-%patch1 -p1
+%patch1 -p1 -b .portability
 sleep 1
 find . \( -name Makefile.in -o -name aclocal.m4 \) -print | xargs touch
 sleep 1
 find . \( -name configure -o -name config.h.in \) -print | xargs touch
 %endif
 
-%patch2 -p1
+%patch2 -p1 -b .robustify
+
+%patch3 -p1 -b .fixes
+
+find . -name \*.sh ! -perm -0100 -print | xargs chmod +x
 
 %build
 # Remove -Wall from default flags.  The makefiles enable enough warnings
@@ -140,13 +174,11 @@
 %endif
 
 %configure CFLAGS="$RPM_OPT_FLAGS -fexceptions"
-make %{?_smp_mflags}
+make -s %{?_smp_mflags}
 
 %install
 rm -rf ${RPM_BUILD_ROOT}
-mkdir -p ${RPM_BUILD_ROOT}%{_prefix}
-
-%makeinstall
+make -s install DESTDIR=${RPM_BUILD_ROOT}
 
 chmod +x ${RPM_BUILD_ROOT}%{_prefix}/%{_lib}/lib*.so*
 chmod +x ${RPM_BUILD_ROOT}%{_prefix}/%{_lib}/elfutils/lib*.so*
@@ -154,16 +186,10 @@
 # XXX Nuke unpackaged files
 { cd ${RPM_BUILD_ROOT}
   rm -f .%{_bindir}/eu-ld
-  rm -f .%{_bindir}/eu-objdump
-  rm -f .%{_includedir}/elfutils/libasm.h
-  rm -f .%{_libdir}/libasm-%{version}.so
-  rm -f .%{_libdir}/libasm.so*
-  rm -f .%{_libdir}/libasm.a
 }
 
 %check
-# XXX elflint not happy on ia64
-make check || :
+make -s check
 
 %clean
 rm -rf ${RPM_BUILD_ROOT}
@@ -180,21 +206,26 @@
 %defattr(-,root,root)
 %doc README TODO
 %{_bindir}/eu-addr2line
+%{_bindir}/eu-ar
 %{_bindir}/eu-elfcmp
 %{_bindir}/eu-elflint
 %{_bindir}/eu-findtextrel
 %{_bindir}/eu-nm
+%{_bindir}/eu-objdump
 %{_bindir}/eu-ranlib
 %{_bindir}/eu-readelf
 %{_bindir}/eu-size
 %{_bindir}/eu-strings
 %{_bindir}/eu-strip
 #%{_bindir}/eu-ld
+%{_bindir}/eu-unstrip
+%{_bindir}/eu-make-debug-archive
 
 %files libs
 %defattr(-,root,root)
+%{_libdir}/libasm-%{version}.so
+%{_libdir}/libasm.so.*
 %{_libdir}/libdw-%{version}.so
-#%{_libdir}/libasm.so.*
 %{_libdir}/libdw.so.*
 %dir %{_libdir}/elfutils
 %{_libdir}/elfutils/lib*.so
@@ -204,15 +235,19 @@
 %{_includedir}/dwarf.h
 %dir %{_includedir}/elfutils
 %{_includedir}/elfutils/elf-knowledge.h
+%{_includedir}/elfutils/libasm.h
 %{_includedir}/elfutils/libebl.h
 %{_includedir}/elfutils/libdw.h
 %{_includedir}/elfutils/libdwfl.h
-#%{_libdir}/libasm.a
 %{_libdir}/libebl.a
-%{_libdir}/libdw.a
-#%{_libdir}/libasm.so
+%{_libdir}/libasm.so
 %{_libdir}/libdw.so
 
+%files devel-static
+%defattr(-,root,root)
+%{_libdir}/libasm.a
+%{_libdir}/libdw.a
+
 %files libelf
 %defattr(-,root,root)
 %{_libdir}/libelf-%{version}.so
@@ -223,10 +258,121 @@
 %{_includedir}/libelf.h
 %{_includedir}/gelf.h
 %{_includedir}/nlist.h
-%{_libdir}/libelf.a
 %{_libdir}/libelf.so
 
+%files libelf-devel-static
+%defattr(-,root,root)
+%{_libdir}/libelf.a
+
 %changelog
+* Wed Oct  1 2008 Roland McGrath <roland@redhat.com> - 0.137-3
+- fix libdwfl regression (#462689)
+
+* Thu Aug 28 2008 Roland McGrath <roland@redhat.com> - 0.137-2
+- Update to 0.137
+  - libdwfl: bug fixes; new segment interfaces;
+             all the libdwfl-based tools now support --core=COREFILE option
+- Resolves: RHBZ #325021, RHBZ #447416
+
+* Mon Jul  7 2008 Tom "spot" Callaway <tcallawa@redhat.com> - 0.135-2
+- fix conditional comparison
+
+* Mon May 12 2008 Roland McGrath <roland@redhat.com> - 0.135-1
+- Update to 0.135
+  - libdwfl: bug fixes
+  - eu-strip: changed handling of ET_REL files wrt symbol tables and relocs
+
+* Wed Apr  9 2008 Roland McGrath <roland@redhat.com> - 0.134-1
+- Update to 0.134
+  - elflint: backend improvements for sparc, alpha (#204170)
+  - libdwfl, libelf: bug fixes (#439344, #438867, #438263, #438190)
+- Remove Conflicts: libelf-devel from elfutils-libelf-devel. (#435742)
+
+* Sun Mar  2 2008 Roland McGrath <roland@redhat.com> - 0.133-2
+- Update to 0.133
+  - readelf, elflint, libebl: SHT_GNU_ATTRIBUTE section handling (readelf -A)
+  - readelf: core note handling for NT_386_TLS, NT_PPC_SPE, Alpha NT_AUXV
+  - libdwfl: bug fixes and optimization in relocation handling
+  - elfcmp: bug fix for non-allocated section handling
+  - ld: implement newer features of binutils linker.
+- Install eu-objdump and libasm, now has limited disassembler support.
+
+* Mon Jan 21 2008 Roland McGrath <roland@redhat.com> - 0.132-3
+- Update to 0.132
+  - libelf: Use loff_t instead of off64_t in libelf.h header. (#377241)
+  - eu-readelf: Fix handling of ET_REL files in archives.
+  - libcpu: Implement x86 and x86-64 disassembler.
+  - libasm: Add interface for disassembler.
+  - all programs: add debugging of branch prediction.
+  - libelf: new function elf_scnshndx.
+
+* Sun Nov 11 2007 Roland McGrath <roland@redhat.com> - 0.131-1
+- Update to 0.131
+  - libdw: DW_FORM_ref_addr support; dwarf_formref entry point now deprecated;
+           bug fixes for oddly-formatted DWARF
+  - libdwfl: bug fixes in offline archive support, symbol table handling;
+             apply partial relocations for dwfl_module_address_section on ET_REL
+  - libebl: powerpc backend support for Altivec registers
+
+* Wed Oct 17 2007 Roland McGrath <roland@redhat.com> - 0.130-3
+- Fix ET_REL support.
+- Fix odd indentation in eu-readelf -x output.
+
+* Tue Oct 16 2007 Roland McGrath <roland@redhat.com> - 0.130-1
+- Update to 0.130
+  - eu-readelf -p option can take an argument like -x for one section
+  - eu-readelf --archive-index (or -c)
+  - eu-readelf -n improved output for core dumps
+  - eu-readelf: handle SHT_NOTE sections without requiring phdrs (#249467)
+  - eu-elflint: ditto
+  - eu-elflint: stricter checks on debug sections
+  - eu-unstrip: new options, --list (or -n), --relocate (or -R)
+  - libelf: new function elf_getdata_rawchunk, replaces gelf_rawchunk;
+            new functions gelf_getnote, gelf_getauxv, gelf_update_auxv
+  - libebl: backend improvements (#324031)
+  - libdwfl: build_id support, new functions for it
+  - libdwfl: dwfl_module_addrsym fixes (#268761, #268981)
+  - libdwfl offline archive support, new script eu-make-debug-archive
+
+* Mon Aug 20 2007 Roland McGrath <roland@redhat.com> - 0.129-2
+- Fix false-positive eu-elflint failure on ppc -mbss-plt binaries.
+
+* Tue Aug 14 2007 Roland McGrath <roland@redhat.com> - 0.129-1
+- Update to 0.129
+  - readelf: new options --hex-dump (or -x), --strings (or -p) (#250973)
+  - addr2line: new option --symbols (or -S)
+  - libdw: dwarf_getscopes fixes (#230235)
+  - libdwfl: dwfl_module_addrsym fixes (#249490)
+
+* Fri Jun  8 2007 Roland McGrath <roland@redhat.com> - 0.128-2
+- Update to 0.128
+  - new program: unstrip
+  - elfcmp: new option --hash-inexact
+- Replace Conflicts: with Provides/Requires using -arch
+
+* Wed Apr 18 2007 Roland McGrath <roland@redhat.com> - 0.127-1
+- Update to 0.127
+  - libdw: new function dwarf_getsrcdirs
+  - libdwfl: new functions dwfl_module_addrsym, dwfl_report_begin_add,
+             dwfl_module_address_section
+
+* Mon Feb  5 2007 Roland McGrath <roland@redhat.com> - 0.126-1
+- Update to 0.126
+  - New program eu-ar.
+  - libdw: fix missing dwarf_getelf (#227206)
+  - libdwfl: dwfl_module_addrname for st_size=0 symbols (#227167, #227231)
+
+* Wed Jan 10 2007 Roland McGrath <roland@redhat.com> - 0.125-3
+- Fix overeager warn_unused_result build failures.
+
+* Wed Jan 10 2007 Roland McGrath <roland@redhat.com> - 0.125-1
+- Update to 0.125
+  - elflint: Compare DT_GNU_HASH tests.
+  - move archives into -static RPMs
+  - libelf, elflint: better support for core file handling
+  - Really fix libdwfl sorting of modules with 64-bit addresses (#220817).
+- Resolves: RHBZ #220817, RHBZ #213792
+
 * Tue Oct 10 2006 Roland McGrath <roland@redhat.com> - 0.124-1
 - eu-strip -f: copy symtab into debuginfo file when relocs use it (#203000)
 - Update to 0.124
